### DISCUSSION TITLE
What can be purpose behind forward declared class without defining and using it anywhere in the project?

### DISCUSSION THREAD

**Discussion by alokmahor** [obsproject/obs-studio#4119] [General]
In the file https://github.com/obsproject/libdshowcapture/blob/master/source/capture-filter.hpp
there is a forward declaration of a class which is nowhere defined and used in whole repo and whole OBS-Studio repo.

`namespace DShow {

// some other declaration

// this class is nowhere defined and used
class CaptureSource;

//some other declaration and definitions
}`

I checked carefully but did not find any definition nor usage of `CaptureSource` class. If I am right then what can be purpose of such declaration?

**Comment by WizardCM** [comment]
A quick look tells me it was added in the initial commit, and didn't end up being used. It happens.
